File: dodam-core/src/main/java/b1nd/dodamcore/member/repository/TeacherRepository.java
package b1nd.dodamcore.member.repository;

import b1nd.dodamcore.member.domain.entity.Member;
import b1nd.dodamcore.member.domain.entity.Teacher;
import b1nd.dodamcore.member.domain.exception.TeacherNotFoundException;
import org.springframework.data.jpa.repository.JpaRepository;
import org.springframework.stereotype.Repository;

import java.util.Optional;

@Repository
public interface TeacherRepository extends JpaRepository<Teacher, Integer> {

    void deleteByMember(Member member);

    Optional<Teacher> findByMember(Member member);

    default Teacher getByMember(Member member) {
        return findByMember(member)
                .orElseThrow(TeacherNotFoundException::new);
    }

}

File: services/service-user/src/main/kotlin/com/b1nd/dodamdodam/user/domain/teacher/repository/TeacherRepository.kt
package com.b1nd.dodamdodam.user.domain.teacher.repository

import com.b1nd.dodamdodam.user.domain.teacher.entity.TeacherEntity
import com.b1nd.dodamdodam.user.domain.user.entity.UserEntity
import org.springframework.data.jpa.repository.JpaRepository

interface TeacherRepository: JpaRepository<TeacherEntity, Long> {
    fun findByUser(user: UserEntity): TeacherEntity?
    fun findAllByUserIn(users: Collection<UserEntity>): List<TeacherEntity>
}